src/Reader.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-multi-assign */
+import S from 'sanctuary';
+import {chain, ap, map, of} from 'fantasy-land';
+import show from 'sanctuary-show';
+
+const $$show = '@@show';
+
+function Reader(run) {
+  if (!(this instanceof Reader)) {
+    return new Reader (run);
+  }
+  this.run = run;
+}
+
+Reader.run = function run(reader, ...args) {
+  return reader.run (...args);
+};
+
+Reader.prototype.chain = Reader.prototype[chain] = function chain_(f) {
+  return new Reader ((r => f (this.run (r)).run (r)));
+};
+
+// ap :: Reader a ~> Reader (a -> b) -> Reader b
+Reader.prototype.ap = Reader.prototype[ap] = function ap_(a) {
+  return a.chain (f => this.map (f));
+};
+
+Reader.prototype.map = Reader.prototype[map] = function map_(f) {
+  return this.chain (a => Reader.of (f (a)));
+};
+
+Reader.prototype.of = Reader.prototype[of] = a => new Reader ((() => a));
+Reader.of = Reader[of] = Reader.prototype.of;
+
+Reader.ask = Reader (S.I);
+
+Reader.prototype.toString = Reader.prototype[$$show] = function show_() {
+  return 'Reader(' + show (this.run) + ')';
```

Comment:
Should we use `Reader (...)` to match Sanctuary style?

Reply:
You mean add a space between `Reader` and the opening parens?

Reply:
To be more specific, change it to `Reader (' + show (this.run) + ')';`?

Reply:
Call be impatient, but I just made the change I assumed you meant.